.github/workflows/publish-cheat-sheet.yml

```diff
@@ -0,0 +1,50 @@
+name: Publish Cheat Sheet
+
+on:
+  push:
+    branches:
+      - main
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Pages
+        uses: actions/configure-pages@v5
+      - name: Build with Jekyll
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./doc/cheat_sheet
+          destination: ./_site
+      - name: Compile cheat sheet for genAI
+        run: sudo python3 doc/cheat_sheet/_scripts/generate_genAI_cheat_sheet.py -d _site
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
```

Comment:
Should we add a condition to skip the deployment if it is a pull-request: `if: github.event_name != 'pull_request'`?

Reply:
It is configured to only run after a push to `main` - I think this would automatically exclude pull requests? https://github.com/amazon-braket/amazon-braket-sdk-python/blob/9fe7082fda51682a0f07461d332fdc6f72b1397d/.github/workflows/publish-cheat-sheet.yml#L3-L6

Reply:
Seems the action is able to be executed from a branch: https://github.com/amazon-braket/amazon-braket-sdk-python/actions/runs/9390972449/job/25959406913

Reply:
Oh, I see, it can be manually executed. Yes, that should be fixed, thanks for catching that.